SingularityUI/app/views/taskHealthcheckNotificationSubview.coffee

```diff
@@ -45,11 +57,19 @@ class taskHealthcheckNotificationSubview extends View
         taskHealthyMessage: taskHealthyMessage
         hasSuccessfulHealthcheck: @model.get('healthcheckResults')?.length > 0 and _.find(@model.get('healthcheckResults'), (item) -> item.statusCode is 200)
         lastHealthcheckFailed: @model.get('healthcheckResults')?.length > 0 and @model.get('healthcheckResults')[0].statusCode isnt 200
+        healthcheckFailureReasonMessage: @healthcheckFailureReasonMessage()
         synced:           @model.synced
         config:           config
         tooManyRetries: @model.get('healthcheckResults').length > maxRetries and maxRetries != 0
         numberFailed: @model.get('healthcheckResults').length
         secondsElapsed: healthTimeoutSeconds
+        doNotDisplayHealthcheckNotification: deployFailureKilledTask
+
+    healthcheckFailureReasonMessage: () ->
```

Comment:
A few things here. First, the method name makes me think we are calculating reasons for any healthcheck failure, when in fact we are only looking for connection refused.

Second thing, the connection refused message isn't very clear. The general confusion is that users see 'connection refused to (slave host):(port)' and assume it's a problem with the slave and don't think to check their own app. We need to clarify that the connection refused was to the app (i.e. the app did not start or wasn't listening on the port). Maybe something more like:

'Healthcheck failed due to a refused connection. It is possible your app did not start properly or was not listening on the anticipated port'